datashader/datashape/predicates.py

```diff
@@ -28,7 +28,7 @@ def isscalar(ds):
     """
     if isinstance(ds, str):
         ds = dshape(ds)
-    if isinstance(ds, DataShape) and len(ds) == 1:
+    elif isinstance(ds, DataShape) and len(ds) == 1:
```

Comment:
Are you sure you can do this? For example, If the calculation above makes ds a DataShape with length 1.

Reply:
You reverted this before I could press the review button, and then it disappeared for me.

Reply:
Yes, I realised I had been a little over-enthusiastic once I got some of the tests working ;)

Reply:
I am very familiar with that feeling.